HopsanGUI/Utilities/GUIUtilities.cpp

```diff
@@ -473,6 +493,25 @@ void reduceVectorSize(QVector<double> &vector, int newSize)
 }
 
 
+//! @brief Limits Y and X vector to specified range in X vector
+//! Both vectors must be of same size!
+//! @param x X-vector
+//! @param y Y-vector
+//! @param min Minimum x value
+//! @param max Maximum x value
+void limitVectorToRange(QVector<double> &x, QVector<double> &y, double min, double max)
+{
+    while(!x.isEmpty() && x.at(0) < min) {
+        x.remove(0);
+        y.remove(0);
```

Comment:
Removing one by one looks inefficient, could use iterators and remove a range, but I am not sure if it matters.
Also min max could be called xmin xmax for clarity
Not super important

Reply:
I improved it so it finds lowest and highest index of range and then uses a single call to remove for the entire length. Also renamed variables to xmin and xmax.